mm2src/trading_api/src/one_inch_api/types.rs

```diff
@@ -0,0 +1,341 @@
+    /// gas limit, in api is a decimal number
+    pub gas: u128,
+}
+
+fn validate_slippage(slippage: f32) -> MmResult<(), ApiClientError> {
+    if !(0.0..=ONE_INCH_MAX_SLIPPAGE).contains(&slippage) {
+        return Err(ApiClientError::OutOfBounds {
+            param: "slippage".to_owned(),
+            value: slippage.to_string(),
+            min: 0.0.to_string(),
+            max: ONE_INCH_MAX_SLIPPAGE.to_string(),
+        }
+        .into());
+    }
+    Ok(())
+}
+
+fn validate_fee(fee: &Option<f32>) -> MmResult<(), ApiClientError> {
+    if let Some(fee) = fee {
+        if !(0.0..=ONE_INCH_MAX_FEE_SHARE).contains(fee) {
+            return Err(ApiClientError::OutOfBounds {
+                param: "fee".to_owned(),
+                value: fee.to_string(),
+                min: 0.0.to_string(),
+                max: ONE_INCH_MAX_FEE_SHARE.to_string(),
+            }
+            .into());
+        }
+    }
+    Ok(())
+}
+
+fn validate_gas_limit(gas_limit: &Option<u128>) -> MmResult<(), ApiClientError> {
+    if let Some(gas_limit) = gas_limit {
+        if gas_limit > &ONE_INCH_MAX_GAS {
+            return Err(ApiClientError::OutOfBounds {
+                param: "gas_limit".to_owned(),
+                value: gas_limit.to_string(),
+                min: 0.to_string(),
+                max: ONE_INCH_MAX_GAS.to_string(),
+            }
+            .into());
+        }
+    }
+    Ok(())
+}
+
+fn validate_parts(parts: &Option<u32>) -> MmResult<(), ApiClientError> {
+    if let Some(parts) = parts {
+        if parts > &ONE_INCH_MAX_PARTS {
+            return Err(ApiClientError::OutOfBounds {
+                param: "parts".to_owned(),
+                value: parts.to_string(),
+                min: 0.to_string(),
+                max: ONE_INCH_MAX_PARTS.to_string(),
+            }
+            .into());
+        }
+    }
+    Ok(())
+}
+
+fn validate_main_route_parts(main_route_parts: &Option<u32>) -> MmResult<(), ApiClientError> {
+    if let Some(main_route_parts) = main_route_parts {
+        if main_route_parts > &ONE_INCH_MAX_MAIN_ROUTE_PARTS {
+            return Err(ApiClientError::OutOfBounds {
+                param: "main route parts".to_owned(),
+                value: main_route_parts.to_string(),
+                min: 0.to_string(),
+                max: ONE_INCH_MAX_MAIN_ROUTE_PARTS.to_string(),
+            }
+            .into());
+        }
+    }
+    Ok(())
+}
+
+fn validate_complexity_level(complexity_level: &Option<u32>) -> MmResult<(), ApiClientError> {
+    if let Some(complexity_level) = complexity_level {
+        if complexity_level > &ONE_INCH_MAX_COMPLEXITY_LEVEL {
+            return Err(ApiClientError::OutOfBounds {
+                param: "complexity level".to_owned(),
+                value: complexity_level.to_string(),
+                min: 0.to_string(),
+                max: ONE_INCH_MAX_COMPLEXITY_LEVEL.to_string(),
+            }
+            .into());
+        }
+    }
+    Ok(())
+}
```

Comment:
you move validation logics inside `fn validate_params` if you don't plan to reuse them. This will also reduce boilerplate code

Reply:
tbh I'd like to have an option to break functions like that (even if those functions are not reused), for code structuring

Reply:
its all good. thanks!